e2e-tests/webdriver-helpers.ts

```diff
@@ -0,0 +1,365 @@
+  }
+
+  async shareSurvey(user: string) {
+    driverInitialized(this.driver);
+    const shareButton = await this.driver.findElement(
+      By.css('button.share-survey-button')
+    );
+    await shareButton.click();
+    await this.waitUntilPresent(By.css('.share-form .email-input input'));
+    await this.enterText(user, By.css('.share-form .email-input input'));
+    const roleSelector = () =>
+      this.driver!.findElement(By.css('.share-form mat-select'));
+    await this.setSelectOption(roleSelector, Role.DATA_COLLECTOR);
+    const doneButton = (
+      await this.driver.findElements(By.css('mat-dialog-container button'))
+    )[1];
+    await doneButton.click();
+    await this.delay();
+    await this.clickContinue();
+  }
+
+  async verifySurveyCreated() {
+    driverInitialized(this.driver);
+    await this.waitUntilPresent(By.css('.job-list-item-container'));
+    const jobElement = await this.driver.findElement(By.css('.job-icon ~ div'));
+    await this.waitUntilTextPresent(jobElement, TestConfig.JOB_NAME);
+  }
+
+  async selectTestSurvey() {
+    driverInitialized(this.driver);
+    const titleElement = await this.findElementByText(By.css('mat-card-title'), TestConfig.SURVEY_TITLE);
+    await titleElement?.click();
+  }
+
+  async waitForSurveySubmissions() {
+    driverInitialized(this.driver);
+    let lastError: Error | null = null;
+    let tries = TestConfig.WAIT_FOR_SUBMISSION_TRIES;
+    do {
+      // Wait for the first LOI submission.
+      try {
+        try {
+          await this.waitUntilPresent(
+            By.css('.loi-icon'),
+            TestConfig.SHORT_TIME_OUT
+          );
+        } catch (e) {
+          const expandButtonSelector = By.css(
+            '.job-list-item-container button'
+          );
+          await this.waitUntilPresent(
+            expandButtonSelector,
+            TestConfig.LONG_TIME_OUT
+          );
+          const expandSubmissionsButton = await this.driver.findElement(
+            expandButtonSelector
+          );
+          await expandSubmissionsButton.click();
+          await this.waitUntilPresent(
+            By.css('.loi-icon'),
+            TestConfig.LONG_TIME_OUT
+          );
+        }
+        // Wait for expected number of submissions.
+        const loiIcon = await this.driver.findElement(By.css('.loi-icon'));
+        await loiIcon.click();
+        await this.waitUntilPresent(By.css('.submission-item'));
+        const submissionItems = await this.driver.findElements(
+          By.css('.submission-item')
+        );
+        if (submissionItems.length < TestConfig.EXPECTED_SUBMISSION_COUNT) {
+          await this.delay(TestConfig.LONG_TIME_OUT);
+          throw new WebDriverHelperException('Not enough survey submissions');
+        }
+        return;
+      } catch (e) {
+        lastError = e as Error;
+      }
+    } while (tries-- > 0);
+    fail(`No survey submissions appeared: ${lastError}`);
+  }
+
+  async verifySurveySubmissions() {
```

Comment:
Submissions exist at the level of job+LOI; to avoid confusion it might be better to just say `verifySubmissions`.

Reply:
Done.